pandas/core/index.py

```diff
@@ -19,3 +21,9 @@
     get_objs_combined_axis,
 )
 from pandas.core.indexes.multi import _sparsify  # noqa:F401
+
+warnings.warn(
+    "pandas.core.index is deprecated and will be removed in a future version.  "
+    "Use pandas.core.indexes.api instead.",
```

Comment:
Can you update this with the same as what you put in the whatsnew? (that is is available from the top-level namespace)